activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/impl/ProcessVariablesPayloadValidator.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.activiti.runtime.api.impl;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.activiti.api.process.model.payloads.ReceiveMessagePayload;
+import org.activiti.api.process.model.payloads.SetProcessVariablesPayload;
+import org.activiti.api.process.model.payloads.SignalPayload;
+import org.activiti.api.process.model.payloads.StartMessagePayload;
+import org.activiti.api.process.model.payloads.StartProcessPayload;
+import org.activiti.common.util.DateFormatterProvider;
+import org.activiti.engine.ActivitiException;
+import org.activiti.spring.process.model.Extension;
+import org.activiti.spring.process.model.ProcessExtensionModel;
+import org.activiti.spring.process.model.VariableDefinition;
+import org.activiti.spring.process.variable.VariableValidationService;
+
+public class ProcessVariablesPayloadValidator  {
+   
+    private final Map<String, ProcessExtensionModel> processExtensionModelMap;
+    private final VariableValidationService variableValidationService;
+    private final DateFormatterProvider dateFormatterProvider;
+    private final VariableNameValidator variableNameValidator;
+ 
+    public ProcessVariablesPayloadValidator(DateFormatterProvider dateFormatterProvider,
```

Comment:
i saw this class before, did we move it from an another package?

Reply:
Yes, the refactoring is made to remove this logic part from RB to Activiti core...
So we have here refactoring + new validation...
The next step will be to remove these classes from RB